src/test/unit/controllers/SummaryOfPublicationsController.test.ts

```diff
@@ -0,0 +1,135 @@
+const onePubJsonData = fs.readFileSync(path.resolve(__dirname, '../mocks/onePublicationJson.json'), 'utf-8');
+const onePubJson = JSON.parse(onePubJsonData);
+const CourtStub = sinon.stub(CourtService.prototype, 'getCourtById');
+const SoPStub = sinon.stub(SummaryOfPublicationsService.prototype, 'getPublications');
+
+describe('Get publications', () => {
+  CourtStub.withArgs(0).resolves(JSON.parse('{"name":"Single Justice Procedure (SJP)"}'));
+  CourtStub.withArgs(1).resolves(JSON.parse('{"name":"New Court"}'));
+  SoPStub.withArgs(0).resolves(sjpCases);
+  SoPStub.withArgs(1).resolves(sjpCases);
+
+  it('should render the Summary of Publications page', async () => {
+
+    const response = {
+      render: () => {
+        return '';
+      },
+    } as unknown as Response;
+
+    const request = mockRequest(i18n);
+    request.query = {courtId: '1'};
+    request.user = {id: 1};
+
+    const responseMock = sinon.mock(response);
+
+    const expectedData = {
+      ...i18n['summary-of-publications'],
+      courtName: 'New Court',
+      publications: sjpCases,
+    };
+
+    responseMock.expects('render').once().withArgs('summary-of-publications', expectedData);
+
+    await publicationController.get(request, response);
+    responseMock.verify();
+  });
+
+  it('should render the SJP if courtId = 0', async () => {
+
+    const response = {
+      render: () => {
+        return '';
+      },
+    } as unknown as Response;
+
+    const request = mockRequest(i18n);
+    request.query = {courtId: '0'};
+    request.user = {id: 1};
+
+    const responseMock = sinon.mock(response);
+
+    const expectedData = {
+      ...i18n['summary-of-publications'],
+      courtName: 'Single Justice Procedure (SJP)',
+      publications: sjpCases,
+    };
+
+    responseMock.expects('render').once().withArgs('summary-of-publications', expectedData);
+
+    await publicationController.get(request, response);
+    responseMock.verify();
+  });
+
+  it('should render the error screen if there is no courtId passed as a param', async () => {
+    const response = {
+      render: () => {
+        return '';
+      },
+    } as unknown as Response;
+    const request = mockRequest(i18n);
+    request.user = {id: 1};
+    const responseMock = sinon.mock(response);
+    responseMock.expects('render').once().withArgs('error');
+  });
+});
+
+describe('Get individual publication and act appropriately', () => {
+
+  it('should open the file directly if only one pub is returned from publicationService', async () => {
+    const response = {
+      redirect: function () {
+        return '';
+      },
+    } as unknown as Response;
+    const request = mockRequest(i18n);
+    request.query = {courtId: '0'};
+    request.user = {id: 1};
+    SoPStub.withArgs(0).resolves(onePub);
+    CourtStub.withArgs('0').resolves(JSON.parse('{"name":"Single Justice Procedure (SJP)"}'));
+    const responseMock = sinon.mock(response);
+    const onePubLength = onePub.length;
```

Comment:
I don't believe we need lines 108 / 109 (The checking of the length), as that is read in from the file rather than the service (Same with test below)